sdk/core/Azure.Core.TestFramework/src/ClientTestFixtureAttribute.cs

```diff
@@ -81,18 +120,22 @@ private void Process(TestSuite testSuite, object serviceVersion, bool isAsync)
                 {
                     foreach (Test parameterizedTest in parameterizedMethodSuite.Tests)
                     {
-                        ProcessTest(serviceVersion, isAsync, serviceVersionNumber, parameterizedTest);
+                        ProcessTest(serviceVersion, isAsync, serviceVersionNumber, parameter, parameterizedTest);
                     }
                 }
                 else
                 {
-                    ProcessTest(serviceVersion, isAsync, serviceVersionNumber, test);
+                    ProcessTest(serviceVersion, isAsync, serviceVersionNumber, parameter, test);
                 }
             }
         }
 
-        private void ProcessTest(object serviceVersion, bool isAsync, int serviceVersionNumber, Test test)
+        private void ProcessTest(object serviceVersion, bool isAsync, int serviceVersionNumber, object parameter, Test test)
         {
+            if (parameter != null)
+            {
+                test.Properties.Set(RecordingDirectoryKey, parameter.ToString());
```

Comment:
RecordingDirectorySuffixKey? As it's not the entire directory name.

Reply:
funny, that's how I initially named it :)